typings/index.d.ts

```diff
@@ -27,6 +27,15 @@ declare namespace commander {
     from: 'node' | 'electron' | 'user';
   }
 
+  // (string & {}) is sort of a hack so we able to show the string types autocompletion
+  // as well as allow any string to be used
+  export type Event =
+  | 'help:header'
+  | 'help:body'
+  | 'help:footer'
+  | (string & {})
```

Comment:
This does indeed seem to work in Visual Studio Code. Do you have any documentation or a source for what `(string & {})` is doing to the type? An intersection of a string and an object with no properties?

Reply:
I went to hunt for some sort of explanation on this but couldn't find any. So I'll try to explain what I believe happens.

An empty interface `{}` is almost like an "any", except it doesn't allow the `undefined | null`, we can see it if you parse the code below on an editor:

```typescript
interface Empty {}

const n:Empty = 0
const s:Empty = 'a'
const a:Empty = []
const o:Empty = {}
const t:Empty = true
const f:Empty = false
const sy:Empty = Symbol()
const u:Empty = undefined //error
const nu:Empty = null //error
```

So the typescript intersection `(string & {})` would mean it has to be a string but can't be null or undefined.

When you declare `const foo: 'a' | 'b' | string = ''` typescript seems to coerce all the values into the `string` type. But when using `(string & {})`, it doesn't, because it is not expecting a "true" string.

The two drawbacks I can find or thing of are:

In some typeguard functions like the one below, it will throw an error, though it is easily bypass-able. 
```typescript
function foo(arg:  'a' | 'b' | (string & {})): 'a' | undefined {
  if (arg === 'a') {
    return arg; // Type '(string & {}) | "a"' is not assignable to type '"a"'
    //return arg as 'a'; // this works
  }
  return
}
```

The other drawback might be that this could be patched in the future, since I couldn't find any documentation or explanation for it. Though I suspect if patched, it would behave just like a `string` type instead. But who knows...

Reply:
An alternative to this would be to overload the "on" function with

```typescript
    on(event: 'help:header' | 'help:body' | 'help:footer' | symbol, listener: (...args: any[]) => void): this;
    on(event: string, listener: (...args: any[]) => void): this;
```